controllers/storagecluster/storageclasses.go

```diff
@@ -528,3 +528,39 @@ func getTopologyConstrainedPools(initData *ocsv1.StorageCluster) string {
 	}
 	return string(topologyConstrainedPoolsStr)
 }
+
+// getTopologyConstrainedPoolsExternalMode constructs the topologyConstrainedPools string for external mode from the data map
+func getTopologyConstrainedPoolsExternalMode(data map[string]string) string {
+	type topologySegment struct {
+		DomainLabel string `json:"domainLabel"`
+		DomainValue string `json:"value"`
+	}
+	// TopologyConstrainedPool stores the pool name and a list of its associated topology domain values.
+	type topologyConstrainedPool struct {
+		PoolName       string            `json:"poolName"`
+		DomainSegments []topologySegment `json:"domainSegments"`
+	}
+	var topologyConstrainedPools []topologyConstrainedPool
+
+	domainLabel := data["topologyFailureDomainLabel"]
+	domainValues := strings.Split(data["topologyFailureDomainValues"], ",")
+	poolNames := strings.Split(data["topologyPools"], ",")
```

Comment:
for debugging add print here and validate what are the values, plus then try to print `topologyConstrainedPools` after the for loop before marshaling